Networking/SimpleImpl/index.md

```diff
@@ -0,0 +1,90 @@
+    // Writes the int into the buf
+    buf.writeInt(toSend);
+  }
+
+  @Override public void fromBytes(ByteBuf buf) {
+    // Reads the int back from the buf. Note that if you have multiple values, you must read in the same order you wrote.
+    toSend = buf.readInt();
+  }
+}
+```
+
+### IMessageHandler
+
+Now, how can we use this packet? Well, first we must have a class that can *handle* this packet. This is created with the `IMessageHandler` interface. Say we wanted to use this integer we sent to give the player that many diamonds on the server. Let's make this handler:
+
+```
+// The params of the IMessageHandler are <REQ, REPLY>, meaning that the first is the packet you are receiving, and the second is the packet you are returning. The returned packet can be used as a "response" from a sent packet.
+public class PacketHandler implements IMessageHandler<Packet, IMessage> {
+  // Do note that the default constructor is required, but implicitly defined in this case
+
+  @Override public IMessage onMessage(Packet message, MessageContext ctx) {
+    // This is the player the packet was sent to the server from
+    EntityPlayerMP serverPlayer = ctx.getServerHandler().playerEntity;
+    // The value that was sent
+    int amount = message.toSend;
+    serverPlayer.inventory.addItemStackToInventory(new ItemStack(Items.diamond, amount));
+    // No response packet
+    return null;
+  }
+}
+```
+It is recommended (but not required) that for organization's sake, this class is an inner class to your Packet class. If this is done, note that the class must also be declared `static`.
+
+### Registering Packets
+So now we have a packet, and a handler for this packet. But the `SimpleNetworkWrapper` needs one more thing to function! For it to use a packet, the packet must be registered with an *discriminator*, which is just an integer used to map packet types between server and client. To do this, call `INSTANCE.registerMessage(PacketHandler.class, Packet.class, 0, Side.Server);` where `INSTANCE` is the `SimpleNetworkWrapper` we defined earlier.
+
+This is quite a complex method, so lets break it down a bit.
+
+- The first parameter is `messageHandler`, which is the class that handles your packet. This class must always have a default constructor, and should have type bound REQ that matches the next argument.
+- The second paramter is `requestMessageType` which is the actual packet class. This class must also have a default constructor and match the REQ type bound of the previous param.
+- The third parameter is the discriminator for the packet. This is a per-channel unique ID for the packet. We recommend you use a static variable to hold the ID, and then call registerMessage using `id++`. This will guarantee 100% unique IDs.
+- The fourth and final parameter is the side that your packet will be ***recieved*** on. If you are planning to send the packet to both sides, it must be registered twice with a ***different*** discriminator.
```

Comment:
I've just had this bit pointed out to me: it doesn't seem to matter if the discriminators are the same for a message class on two sides. Is this actually important or not?